engine-nextgen/src/main/kotlin/graphql/nadel/enginekt/blueprint/IntrospectionService.kt

```diff
@@ -0,0 +1,29 @@
+package graphql.nadel.enginekt.blueprint
+
+import graphql.ExecutionInput
+import graphql.GraphQL
+import graphql.GraphqlErrorHelper.toSpecification
+import graphql.language.AstPrinter
+import graphql.nadel.Service
+import graphql.nadel.ServiceExecution
+import graphql.nadel.ServiceExecutionResult
+import graphql.schema.GraphQLSchema
+
+internal class IntrospectionService constructor(schema: GraphQLSchema) :
+    Service(name, schema, makeServiceExecution(schema), null, null) {
+
+    companion object {
+        const val name = "__introspection"
+
+        private fun makeServiceExecution(schema: GraphQLSchema): ServiceExecution {
+            return ServiceExecution { params ->
+                val executionInput = ExecutionInput.newExecutionInput()
+                    .query(AstPrinter.printAstCompact(params.query))
```

Comment:
Future could look like this: we can extend the ServiceExecution to take actually a ExecutableNormalizedField which then could be used in the future to execute the request. I want that future, but this is not immediate as we we need an engine being able to execute a normalized fields instead of Document/String.